### ISSUE TITLE
elm-format needs strange brackets to work

### ISSUE THREAD

**Issue by Shaomada** [avh4/elm-format#211]
Today I came across a case where elm-format needs some additional brackets for constuctors from imported modules when pattern matching:

```elm
module Submodule exposing (..)


type B
    = B
```
```elm
module Module exposing (..)

import Submodule
import Html


type A
    = A Submodule.B


main : Html.Html a
main =
    case A Submodule.B of
        A (Submodule.B) ->
            Html.text "Hello, World!"

```
I'm have a strong feeling the pair of brackets around ```Submodule.B``` in the second to last line should not be necessary, but without them elm-format fails with the error message:
```
-- SYNTAX PROBLEM ----------------------------------------------- src/Module.elm

I ran into something unexpected when parsing your code!

14│         A Submodule.B ->
                       ^
I am looking for one of the following things:

    ->
    a cons operator '::'
    a pattern
    reserved word `as`
    whitespace
```
elm-make and elm-reactor work fine without the brackets.

**Comment by avh4** [comment]
Hello, the parens were required in Elm 0.16, but not in Elm 0.17.  this is the same issue as https://github.com/avh4/elm-format/issues/184 and will be fixed in the next release.

**Comment by avh4** [comment]
This should be fixed in 0.4.0-alpha https://github.com/avh4/elm-format#installation-
